sn_networking/src/record_store.rs

```diff
@@ -321,19 +350,28 @@ impl RecordStore for NodeRecordStore {
             return Err(Error::ValueTooLarge);
         }
 
-        if self.records.contains(&record.key) {
-            trace!(
-                "Unverified Record {:?} already exists.",
-                PrettyPrintRecordKey::from(&record.key)
-            );
+        let record_key = PrettyPrintRecordKey::from(&record.key);
 
-            // Blindly sent to validation to allow double spend can be detected.
-            // TODO: consider avoid throw duplicated chunk to validation.
+        // Chunk with existing key do not to be stored again.
+        // `Spend` or `Register` with same content_hash do not to be stored again,
```

Comment:
`// `Spend` or `Register` with same content_hash do not to be stored again,`

`do need to be stored` no?

Reply:
that's for the case of `same key AND same content_cash`, so `do not need to be stored again`